src/PlatformLinux.cpp

```diff
@@ -183,6 +183,24 @@ bool Platform::dirRemove(const std::string& path) {
 	return true;
 }
 
+std::string Platform::dirGetLocal() {
+    char abs_path[1024];
+    ssize_t len = readlink("/proc/self/exe", abs_path, 1024);
+    if(len < 0|| len >= 1024)  {
+        return "./";
+    }
+    //remove executable name from abs_path
+    char break_point = '/';
+    char break_string = '\0';
+    for(ssize_t i = len; i >= 0; --i) {
+        if(abs_path[i] == break_point) {
```

Comment:
What happens if the path separator is never found?  Also same comment on whitespaces as above.

Reply:
If the program starts i believe it should be in a location. Even if it is in the root dir the abs_path will contain at least "/flare", so it has "/" in it.
readlink() on success returns the number of bytes placed in abs_path, on error, -1 is returned [as per man page](https://man7.org/linux/man-pages/man2/readlink.2.html). This is checked with the if statement and covers all errors generated when calling readlink();

Reply:
`/proc/[pid]/exe [isn't seemed to be warrantied to be absolute](https://man7.org/linux/man-pages/man5/proc.5.html).  I suppose most of the cases there must be a slash somewhere in the returned path, so this might be safely ignored.